### PR TITLE
ec-332 Explorer - create route and view: /starting block date/finish block date

### PR DESCRIPTION
**Pull request by GusevTimofey**

### CODE REVIEW

File: app/controllers/HistoryController.scala [hunk 1]
```diff
@@ -97,4 +102,30 @@ class HistoryController @Inject()(cc: ControllerComponents, historyService: Hist
     }
   }
 
+  def listHeadersByDateFromToView(date1: String, date2: String): Action[AnyContent] = Action.async {
```
Comment (GRYE) [inline app/controllers/HistoryController.scala]: Maybe name arguments fromDate and toDate?

File: conf/routes [hunk 2]
```diff
@@ -34,4 +34,6 @@ GET        /api/transactions/block/:modifierId/outputs/unspent           control
 GET        /api/transactions/tx/range/:from/:to                          controllers.TransactionsController.findTransactionByBlockHeightRangeApi(from: Int, to: Int)
 
 GET        /api/block/:id                                                controllers.BlockController.findBlockApi(id: String)
-GET        /block/:id                                                    controllers.BlockController.findBlockView(id: String)
\ No newline at end of file
+GET        /block/:id                                                    controllers.BlockController.findBlockView(id: String)
+GET        /findHeadersByDate/:date/:count                               controllers.HistoryController.findHeadersByDateView(date: String, count: Int)
+GET        /api/findHeadersByDate/:date/:count                           controllers.HistoryController.findHeadersByDateApi(date: String, count: Int)
```
Comment (jhnsmth) [inline conf/routes]: this methods should have `/api/history/headers/` prefix

File: app/utils/CheckAction.scala [hunk 3]
```diff
@@ -79,6 +79,7 @@ class FromCountCheckActionFactory @Inject()(parser: BodyParsers.Default) {
 
 class DateFromCountAction(parser: BodyParsers.Default, date: String, count: Int) extends ActionBuilderImpl(parser) {
   private val sdf: SimpleDateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss")
+  sdf.setLenient(true)
```
Comment (jhnsmth) [inline app/utils/CheckAction.scala]: `true` is the default value, so it can be removed

Comment (jhnsmth) [reply app/utils/CheckAction.scala]: sry, misunderstood

File: app/utils/CheckAction.scala [hunk 4]
```diff
@@ -105,4 +106,23 @@ class Base58CheckAction(parser: BodyParsers.Default, address: String) extends Ac
...
+}
+
+class DateFromToCheckAction(parser: BodyParsers.Default, fromDate: String, toDate: String) extends ActionBuilderImpl(parser) {
+  private val sdf: SimpleDateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss")
+  sdf.setLenient(true)
+
+  override def invokeBlock[A](request: Request[A], block: Request[A] => Future[Result]): Future[Result] = {
+    val fromDateParse: Try[Date] = Try(sdf.parse(fromDate + " 0:0:0"))
+    val toDateParse: Try[Date] = Try(sdf.parse(toDate + " 23:59:59"))
+    if (fromDateParse.map(from => toDateParse.map(_.getTime >= from.getTime).getOrElse(false)).getOrElse(false))
```
Comment (GRYE) [inline app/utils/CheckAction.scala]: You can change condition to `fromDateParse.flatMap(from => toDateParse.map(_.getTime >= from.getTime)).getOrElse(false)`

File: app/utils/CheckAction.scala [hunk 5]
```diff
@@ -105,4 +106,23 @@ class Base58CheckAction(parser: BodyParsers.Default, address: String) extends Ac
 
 class Base58CheckActionFactory @Inject()(parser: BodyParsers.Default) {
   def apply(address: String): Base58CheckAction = new Base58CheckAction(parser, address)
+}
+
+class DateFromToCheckAction(parser: BodyParsers.Default, fromDate: String, toDate: String) extends ActionBuilderImpl(parser) {
+  private val sdf: SimpleDateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss")
```
Comment (GRYE) [inline app/utils/CheckAction.scala]: You have 2 exactly the same instances of SimpleDateFormat, maybe you should reuse them

**Pull request merged**